server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByEngineDataSet.java

```diff
@@ -66,6 +66,83 @@ public GroupByEngineDataSet(QueryContext context, GroupByTimePlan groupByTimePla
     initGroupByEngineDataSetFields(context, groupByTimePlan);
   }
 
+  protected Pair<Long, Long> getFirstTimeRange() {
+    long retEndTime;
+    if (isIntervalByMonth) {
+      // calculate interval length by natural month based on startTime
+      // ie. startTIme = 1/31, interval = 1mo, curEndTime will be set to 2/29
+      retEndTime = Math.min(calcIntervalByMonth(interval), endTime);
+    } else {
+      retEndTime = Math.min(startTime + interval, endTime);
+    }
+    return new Pair<>(startTime, retEndTime);
+  }
+
+  protected Pair<Long, Long> getLastTimeRange() {
+    long retStartTime, retEndTime;
```

Comment:
Coding style issue. Please do not define multiple variables in a single line.

Reply:
done.